### PR TITLE
feat(zk_toolbox): Add contract verifier support for zk toolbox

### PR DESCRIPTION
**Pull request by matias-gonz**
## What ❔
Add contract verifier support for zk toolbox

### CODE REVIEW

File: zk_toolbox/crates/zk_inception/src/commands/contract_verifier/args/init.rs [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+}
+
+#[derive(Debug, Clone, ValueEnum, strum::EnumIter, PartialEq, Eq, Copy)]
+pub enum Arch {
+    LinuxAmd,
+    LinuxArm,
+    MacosAmd,
+    MacosArm,
+    WindowsAmd,
+}
```
Comment (Deniallugo) [inline zk_toolbox/crates/zk_inception/src/commands/contract_verifier/args/init.rs:44]: should we simply use https://doc.rust-lang.org/std/env/consts/constant.OS.html and https://doc.rust-lang.org/std/env/consts/constant.ARCH.html

File: zk_toolbox/crates/zk_inception/src/commands/contract_verifier/args/init.rs [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+                .context(MSG_GET_ZKSOLC_RELEASES_ERR)?
+                .to_string();
+            releases.push(ZkSolcVersion {
+                version: version.clone(),
+                arch,
+                url,
+            });
+        }
+    }
+
```
Comment (Deniallugo) [inline zk_toolbox/crates/zk_inception/src/commands/contract_verifier/args/init.rs:153]: could you simply write a desired struct and implement serialize/deserialize?

File: zk_toolbox/crates/zk_inception/src/commands/contract_verifier/init.rs [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    let link_to_code = ecosystem.link_to_code;
+    let zksolc_path = link_to_code
+        .join("etc/zksolc-bin/")
+        .join(args.zksolc_version.version);
+    let zkvyper_path = link_to_code
+        .join("etc/zkvyper-bin/")
+        .join(args.zkvyper_version.version);
+    let vyper_path = link_to_code
+        .join("etc/vyper-bin/")
+        .join(args.vyper_version.version.replace("v", ""));
```
Comment (Deniallugo) [inline zk_toolbox/crates/zk_inception/src/commands/contract_verifier/init.rs:22]: is it required to be inside the repo?

File: zk_toolbox/crates/zk_inception/src/commands/contract_verifier/init.rs [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    if shell.path_exists(binary_path.clone()) {
+        logger::info(&msg_binary_already_exists(name));
+        return Ok(());
+    }
+
+    let spinner = Spinner::new(&msg_downloading_binary_spinner(name));
+    Cmd::new(cmd!(shell, "mkdir -p {path}")).run()?;
+    Cmd::new(cmd!(shell, "wget {url} -O {binary_path}")).run()?;
+    Cmd::new(cmd!(shell, "chmod +x {binary_path}")).run()?;
+    spinner.finish();
```
Comment (Deniallugo) [inline zk_toolbox/crates/zk_inception/src/commands/contract_verifier/init.rs:42]: is it the only binary? 
i think we have to choose minimal version and download everything above

File: zk_toolbox/crates/zk_inception/src/commands/contract_verifier/init.rs (hunk 3 above)
Comment (matias-gonz) [reply zk_toolbox/crates/zk_inception/src/commands/contract_verifier/init.rs:22]: Yes, the contract verifier uses the binaries inside the repo:
https://github.com/matter-labs/zksync-era/blob/main/core/bin/contract-verifier/src/main.rs#L21

**Review by Deniallugo — APPROVED** [review]

**Review by Deniallugo — DISMISSED** [review]

**Review by perekopskiy — DISMISSED** [review]

**Review by Deniallugo — APPROVED** [review]

**Pull request merged**